dashboard/scripts/update_under_13_students.rb

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env ruby
+# Update all students under age 13, or with age not set, to have sharing_disabled = true.
+# sharing_disabled is contained in properties of the User model
+require_relative('../config/environment')
+
+CDO.log.info "Starting to batch update all students under 13"
+number_under_13 = 0
+
+User.find_in_batches do |users_batch|
+  ActiveRecord::Base.transaction do
```

Comment:
Why containing each batch in a transaction? Doing that will introduce table locks that might have side effects, in opposition to description. Unless there's a compelling reason for transactions, we should instead update each row individually.

Given that, if we don't care about specific batches, we can simplify this to use `find_each` instead of `find_in_batches`, which is effectively the same for the inner loop.

Most importantly, it appears we can greatly improve this with a sql query: [under_13?](https://github.com/code-dot-org/code-dot-org/blob/fba0833d62b24335593c014c8a8d14e3314a9f03/dashboard/app/models/user.rb#L1014) calculates the age from `birthday` which is a DB column. We could calculate the min birthday required to be 13 or over and query by that date, plus null (which apparently represents < 4 years old):

```ruby
min_birthday = Date.today - 13.years
User.where('birthday IS NULL OR birthday < ?', min_birthday).find_each ...
```

And given we can do that in sql, why iterate at all? Why not simply update all the rows at once:

```ruby
min_birthday = Date.today - 13.years
User.
  where('birthday IS NULL OR birthday < ?', min_birthday).
  update_all(sharing_disabled: true)
```

If you want to log the count, you can run the query twice: once for update and once for count:
```ruby
min_birthday = Date.today - 13.years
users = User.where('birthday IS NULL OR birthday < ?', min_birthday)
users.update_all sharing_disabled: true

puts "Finished updating #{users.count} students"
```

Reply:
I'd love it to be as simple as the last two, but `sharing_disabled` isn't a column, it's on properties, so I'll modify to something similar to the first suggestion.